play-services-core/src/main/kotlin/org/microg/gms/gcm/PushRegisterService.kt

```diff
@@ -78,7 +78,21 @@ private suspend fun ensureAppRegistrationAllowed(context: Context, database: Gcm
 }
 
 suspend fun completeRegisterRequest(context: Context, database: GcmDatabase, request: RegisterRequest, requestId: String? = null): Bundle = suspendCoroutine { continuation ->
-    PushRegisterManager.completeRegisterRequest(context, database, requestId, request) { continuation.resume(it) }
+    PushRegisterManager.completeRegisterRequest(context, database, requestId, request) {
+        val errorMsg = it.getString(EXTRA_ERROR)
+        Log.w(TAG, "completeRegisterRequest error: $errorMsg")
+        if (errorMsg == ERROR_INVALID_FID && !request.delete) {
```

Comment:
If `requestId` is non-null, the value in the error field will be prefixed with `|ID|$requestId|`, so you should compare errorMsg against `PushRegisterManager.attachRequestId(ERROR_INVALID_FID, requestId)`